test/marketplace/MarketPlace.sol

```diff
@@ -98,12 +98,25 @@ contract MarketPlace {
     return true;
   }
 
+  /// @notice Can be called at any point by the admin to pause markets and prevent any further transactions
+  function pauseMarkets() public circuitBreaker() onlyAddress(admin) returns (bool) {
+    isPaused = true;
+    return true;
+  }
+
+  /// @notice Can be called at any point by the admin to unpause markets and resume any further transactions
+  function resumeMarkets() public onlyAddress(admin) returns (bool) {
+    require(isPaused == true, 'markets not paused');
+    isPaused = false;
+    return true;
+  }
+
   /// @notice Allows Swivel caller to deposit their underlying, in the process splitting it - minting both zcTokens and vault notional.
   /// @param u Underlying token address associated with the market
   /// @param m Maturity timestamp of the market
   /// @param t Address of the depositing user
   /// @param a Amount of notional being added
-  function mintZcTokenAddingNotional(address u, uint256 m, address t, uint256 a) external onlyAddress(swivel) returns (bool) {
+  function mintZcTokenAddingNotional(address u, uint256 m, address t, uint256 a) external circuitBreaker() onlyAddress(swivel) returns (bool) {
```

Comment:
no. bad naming all round.

we've talked about the need for a circuit breaker and that's cool. we aren't calling it that. we can slim it down to just the interval var `paused` and a modifier `unpaused` that needs to return false. obvs this can be inverted to an internal var `unpaused` and a modifier `paused` ... whichever way we want to go